torch/lib/c10d/ProcessGroup.hpp

```diff
@@ -222,14 +222,14 @@ class ProcessGroup : public torch::CustomClassHolder {
   // Gathers a single tensor inputBuffer into a single buffer outputBuffer that
   // is interpreted as a contigious collection of size inputBuffer * WORLD_SIZE.
   // For implementers of ProcessGroup API and advanced users only.
-  virtual c10::intrusive_ptr<ProcessGroup::Work> allgather_base(
+  virtual c10::intrusive_ptr<ProcessGroup::Work> _allgather_base(
```

Comment:
Do we need to rename allgather_base to _allgather_base? E.g. We could just leave allgather_base ASIS, but  name it _all_gather_base in py API to mark it "for private use".

Reply:
discussed offline. I think it's fine to rename it here, let's add the corresponding comment so users have a better idea re: future of this function?